snapd-glib/snapd-client-sync.c

```diff
@@ -1964,3 +1964,53 @@ snapd_client_get_notices_with_filters_sync (SnapdClient *self,
     end_sync (&data);
     return snapd_client_get_notices_with_filters_finish (self, data.result, error);
 }
+
+/**
+ * snapd_client_get_model_assertion_sync:
+ * @client: a #SnapdClient.
+ * @cancellable: (allow-none): a #GCancellable or %NULL.
+ * @error: (allow-none): #GError location to store the error occurring, or %NULL to ignore.
+ *
+ * Synchronously get the active model assertion for this device.
+ *
+ * Returns: an assertion or %NULL on error.
```

Comment:
Shouldn't it include `(transfer full) (allow-none)`?

And the same in the other three methods (`get_model_assertion_finish`, `get_serial_assertion_sync` and `get_serial_assertion_finish`).

Reply:
Also, why is it an "assertion"? Is it because it contains JSON data instead of the serial value itself?

Reply:
I think it's not supposed to have `allow-none`, since the function only returns `NULL` on an error. In a nullable language the API user isn't supposed to handle the `NULL` case, this only occurs during an exception.

It's not "string" because the gchar * already implies it's a string. The string is a textual description of an assertion. I originally had it return `SnapdAssertion` but the existing assertion APIs use strings described like this, so I did it the same for consistency.